onboard-server/src/main/java/no/vegvesen/ixn/serviceprovider/OnboardRestController.java

```diff
@@ -289,13 +292,17 @@ public PrivateChannelApi addPrivateChannel(@PathVariable String serviceProviderN
 			throw new PrivateChannelException("Client channel cannot be null");
 		}
 
-		ServiceProvider serviceProviderToUpdate = getOrCreateServiceProvider(serviceProviderName);
-
-		PrivateChannel newPrivateChannel = serviceProviderToUpdate.addPrivateChannel(clientChannel.getPeerName());
+		boolean channelExists = privateChannelRepository.existsByPeerNameAndServiceProviderName(clientChannel.getPeerName(), serviceProviderName);
```

Comment:
Remove, the only unique combination should be serviceProviderName + peerName + queueName